src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1055,6 +1032,37 @@ internal void OnCloseTimeOutTimerElapsed(object source)
         }
 
         #endregion
+    
+        #region Protected Methods
+
+        protected void DisposeMessageQueue()
+        {
+            // Stop session processing thread.
+            try
+            {
+                _sessionMessageQueue.CompleteAdding();
+            }
+            catch (ObjectDisposedException)
+            {
+                // Object already disposed.
+            }
+
+            _sessionMessageQueue.Dispose();
```

Comment:
should this be up in the `try`?

Reply:
No, Dispose() is idempotent.  Other object methods however can throw the `ObjectDisposedException`, so require the try/catch.